SCClassLibrary/External/Ableton/LinkClock.sc

```diff
@@ -11,6 +13,169 @@ LinkClock : TempoClock {
 		).prInitFromTempoClock(clock)
 	}
 
+	init { |tempo, beats, seconds, queueSize|
+		super.init(tempo, beats, seconds, queueSize);
+		id = 0x7FFFFFFF.rand;  // to ignore 'self' queries
```

Comment:
That's a good point (though I'd suggest, in a typical use case, that users should initialize clocks across the network before pulling random numbers for musical purposes).

I can also imagine an ensemble may just assign each member an id that they use every time, so, making it configurable is a good idea.

Between configuring this and network ports (which aren't a part of Link's protocol, nor of normal clock operation), I'm thinking it would be better to split meter sync into a separate class (to avoid god-object design). So instead of `LinkClock.new.syncMeter_(true)`, you might do:

```
l = LinkClock.new etc;
m = SCLinkClockMeterWatcher(l);
```

... and address commands to the separate sync object. I think it makes more sense than cramming tangentially related options into LinkClock itself.

Reply:
> I'm thinking it would be better to split meter sync into a separate class (to avoid god-object design)

yes, that's a great idea! i hadn't even considered that

Reply:
Ok, I'll try to do that next week, and incorporate the other recommendations at that time. (I generally agree with the other review comments, won't reply to them all individually but I'm noting them to include in v2.)

Reply:
sounds good! fyi, when there are a lot of comments on a PR i find using the :+1: reaction and/or "resolve conversation" buttons helpful to keep track of what has and hasn't been dealt with yet.

Reply:
(sorry if you already do that, i've seen some people not do it and with PRs this big it can generate quite the sprawl)